### DISCUSSION TITLE
Running tests via pytest throws  RuntimeError: Event loop is closed

### DISCUSSION THREAD

**Discussion by FumaxIN** [fastapi/fastapi#11556] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
client = TestClient(app)


def test_login():
    response = client.post(
        "/login",
        data={
            "username": "testuser1",
            "password": "qwerty@123"
        }
    )
    assert response.status_code == 200
    assert response.json()["access_token"]


def test_login_b():
    response = client.post(
        "/login",
        data={
            "username": "johndoe",
            "password": "qwerty@123"
        }
    )
    assert response.status_code == 200
    assert response.json()["access_token"]


### LOGIN ###

@app.post("/login", tags=["auth"])
async def login(request: OAuth2PasswordRequestForm = Depends()):
    user = await db["users"].find_one({"username": request.username})
    if not user:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Invalid username or password"
        )
    if not Hash.verify(user["password"], request.password):
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"Invalid username or password"
        )
    access_token = create_access_token(data={"sub": user["username"]})
    return {"access_token": access_token, "token_type": "bearer"}
```

### Description

I am working on a blog management project using MongoDB Atlas as my database.

All the routes are working fine but I am facing issues with Tests.
The first test always works fine but the second and other subsequent tests fail with an error `RuntimeError: Event loop is closed`

Please help me understand and fix this.

### Operating System

Linux

### FastAPI Version

0.111.0

### Pydantic Version

2.7.1

### Python Version

3.12.3

**Comment by YuriiMotov** [comment]
I guess you are using pytest-asyncio library.
If so, it's a known issue of pytest-asyncio (https://github.com/pytest-dev/pytest-asyncio/issues/706). It's recommended to use version 0.21.2 or less of pytest-asyncio to avoid this problem

**Comment by FumaxIN** [comment, marked as answer]
This [StackOverflow answer](https://stackoverflow.com/a/73811894) did the job

**Reply by phvmous** [reply]
Thank you so much for posting this solution.
